CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+cmake_minimum_required(VERSION 3.20)
+
+project(wxcharts
+        VERSION 1.1
+        LANGUAGES CXX)
+
+################
+# config cmake #
+################
+
+set(CMAKE_CXX_STANDARD 17 CACHE STRING "C++ Standard")
+set(CMAKE_CXX_STANDARD_REQUIRED ON CACHE BOOL "")
+set(CMAKE_CXX_EXTENSIONS OFF CACHE BOOL "")
+set(CMAKE_DEBUG_POSTFIX "-d" CACHE STRING "Postfix for debug output")
+set(wxWidgets_ROOT_DIR "" CACHE PATH "")
+set(wxWidgets_wxrc_EXECUTABLE "" CACHE FILEPATH "")
+
+
+option(BUILD_SHARED_LIBS "Build shared libraries" ON)
```

Comment:
The default build should be a static build.

Reply:
This is because at this stage we can't really guarantee any compatibility between versions. If developers want a shared library we shouldn't stop them but we can't guarantee compatibility between versions,

Reply:
What about a warning for a shared build?

Reply:
Setting the default to a static build and providing an option to make a shared build that prints a warning saying that API can change would be perfect.

Reply:
Fine